_posts/2020-04-07-trying-out-julia.md

```diff
@@ -0,0 +1,227 @@
+
+![Julia REPL](/img/trying-out-julia/julia-repl.png)
+
+Great we have a fancy calculator which seems to be even calculating correctly! After some 20-20 hindsight I can say there is some important information already displayed in the welcome ASCII-art, mainly the ]-mark. This is the way to access Julia package manager for adding dependencies. Also with hindsight using `activate <env_name>` with the package manager allows you to create python like virtual environments to contain your packages within your project.
+
+So good so far but I'm one of those guys who likes IDEs especially when learning new languages. The Julia homepage has a nice list of supported and familiar editors and IDEs. But why use something you already know? Let's go all in with new stuff and select Atom editor with JUNO plug-in. Ok yes part of the reason selecting Juno was that it looked cool in screenshots at [junolab.org](https://junolab.org). The installation was quite straight forward as Atom had a .deb-package and after installing that I just needed to add the uber-juno plug-in with atom's built in tools. Yes now I had all the tools in place, time to start coding.
+
+## What to do
+
+As said I have some background with MatLab and wanted to do some visualizations as they seem to be the strong point of Julia. So what data to visualize, something current...surprise surprise let's use COVID-19 data. Basic status data can be acquired from [api.covid19api.com](https://api.covid19api.com/) using REST calls. So basically to get day to day confirmed cases from Finland issue GET call to [api.covid19api.com/country/finland/status/confirmed](https://api.covid19api.com/country/finland/status/confirmed) and you get an array of the following json objects
+
+```json
+{
+    "Country": "Finland",
+    "Province": "",
+    "Lat": 61.9241,
+    "Lon": 25.7482,
+    "Date": "2020-03-31T00:00:00Z",
+    "Cases": 1418,
+    "Status": "confirmed"
+},
+```
+
+After some document browsing I know I need to add some packages to my Julia environment 
+
+| Package    | Purpose                                           |
+| :--------- | ------------------------------------------------- |
+| HTTP       | To make HTTP calls                                |
+| JSON       | Json tools for e.g. parsing                       |
+| Plots      | Basic plotting support                            |
+| StatsPlots | Extended plotting support e.g. dataframes support |
+
+This is achieved with the package manager in REPL by first issuing the ]-mark and then just `"activate <env_name>"` and then `"add HTTP"` and so on. Quite handy, I especially like that the package manager is baked into the REPL and not a completely separate tool. Now for some basic coding
```

Comment:
the REPL sanoisin

] ei ole varmaan mark? Character?

This is quite handy and I especially... baked into the REPL instead of being a completely separate tool.